core/src/warm_quic_cache_service.rs

```diff
@@ -36,23 +36,25 @@ impl WarmQuicCacheService {
                 let slot_jitter = thread_rng().gen_range(-CACHE_JITTER_SLOT, CACHE_JITTER_SLOT);
                 let mut maybe_last_leader = None;
                 while !exit.load(Ordering::Relaxed) {
-                    let leader_pubkey =  poh_recorder
-                        .lock()
-                        .unwrap()
-                        .leader_after_n_slots((CACHE_OFFSET_SLOT + slot_jitter) as u64);
-                    if let Some(leader_pubkey) = leader_pubkey {
-                        if maybe_last_leader
-                            .map_or(true, |last_leader| last_leader != leader_pubkey)
-                        {
-                            maybe_last_leader = Some(leader_pubkey);
-                            if let Some(addr) = cluster_info
-                                .lookup_contact_info(&leader_pubkey, |leader| leader.tpu)
+                    if use_quic() {
```

Comment:
can we just not start the service instead?

Reply:
Tried doing that initially. That meant plumbing `use_quic` flag from `validator/src/main.rs` to `validator.rs` to `tvu.rs`. Also, this change enabled somewhat dynamic control over when `quic` actually gets enabled.

Reply:
Updated the PR to not start the warmup service if quic is not enabled.